components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -97,9 +97,104 @@ func InjectReconciliationLock(meta *metav1.ObjectMeta) error {
 	return nil
 }
 
+// resourceConfig holds parsed and validated OAuth proxy resource configuration
+type resourceConfig struct {
+	cpuRequest    resource.Quantity
+	memoryRequest resource.Quantity
+	cpuLimit      resource.Quantity
+	memoryLimit   resource.Quantity
+}
+
+// parseAndValidateAuthSidecarResources parses OAuth proxy resource annotations, applies defaults,
+// and validates that requests <= limits. This centralizes all resource parsing and validation logic.
+func parseAndValidateAuthSidecarResources(notebook *nbv1.Notebook) (*resourceConfig, error) {
+	// Default values
+	defaultCPURequest := resource.MustParse(DefaultAuthSidecarCPURequest)
+	defaultMemoryRequest := resource.MustParse(DefaultAuthSidecarMemoryRequest)
+	defaultCPULimit := resource.MustParse(DefaultAuthSidecarCPULimit)
+	defaultMemoryLimit := resource.MustParse(DefaultAuthSidecarMemoryLimit)
+
+	// Start with defaults
+	config := &resourceConfig{
+		cpuRequest:    defaultCPURequest,
+		memoryRequest: defaultMemoryRequest,
+		cpuLimit:      defaultCPULimit,
+		memoryLimit:   defaultMemoryLimit,
+	}
+
+	annotations := notebook.GetAnnotations()
+	if annotations == nil {
+		return config, nil
+	}
+
+	// Parse and validate each annotation, updating config if valid
+	if cpuReqStr := annotations[AnnotationAuthSidecarCPURequest]; cpuReqStr != "" {
+		cpuReq, err := resource.ParseQuantity(strings.TrimSpace(cpuReqStr))
+		if err != nil {
+			return nil, fmt.Errorf("invalid CPU request value '%s': %w", cpuReqStr, err)
+		}
+		config.cpuRequest = cpuReq
+	}
+
+	if memReqStr := annotations[AnnotationAuthSidecarMemoryRequest]; memReqStr != "" {
+		memReq, err := resource.ParseQuantity(strings.TrimSpace(memReqStr))
+		if err != nil {
+			return nil, fmt.Errorf("invalid memory request value '%s': %w", memReqStr, err)
+		}
+		config.memoryRequest = memReq
+	}
+
+	if cpuLimitStr := annotations[AnnotationAuthSidecarCPULimit]; cpuLimitStr != "" {
+		cpuLimit, err := resource.ParseQuantity(strings.TrimSpace(cpuLimitStr))
+		if err != nil {
+			return nil, fmt.Errorf("invalid CPU limit value '%s': %w", cpuLimitStr, err)
+		}
+		config.cpuLimit = cpuLimit
+	}
+
+	if memLimitStr := annotations[AnnotationAuthSidecarMemoryLimit]; memLimitStr != "" {
+		memLimit, err := resource.ParseQuantity(strings.TrimSpace(memLimitStr))
+		if err != nil {
+			return nil, fmt.Errorf("invalid memory limit value '%s': %w", memLimitStr, err)
+		}
+		config.memoryLimit = memLimit
+	}
```

Comment:
as this repetative, shall we use a map and loop through that or convert this into a separate func ?

something like: 
```suggestion
	resourceMap := map[string]*resource.Quantity{
		AnnotationAuthSidecarCPURequest:    &config.cpuRequest,
		AnnotationAuthSidecarMemoryRequest: &config.memoryRequest,
		AnnotationAuthSidecarCPULimit:      &config.cpuLimit,
		AnnotationAuthSidecarMemoryLimit:   &config.memoryLimit,
	}

	// Parse annotation from resourceMap and update config
	for key, quantity := range resourceMap {
		if valStr, ok := annotations[key]; ok && valStr != "" {
			val, err := resource.ParseQuantity(strings.TrimSpace(valStr))
			if err != nil {
				return nil, fmt.Errorf("invalid value for annotation '%s': '%s': %w", key, valStr, err)
			}
			*quantity = val
		}
	}
```
Note: Not tested